src/components/MessageList/hooks/useMarkRead.ts

```diff
@@ -1,17 +1,27 @@
-import { useEffect, useRef } from 'react';
+import { useEffect } from 'react';
 import {
   StreamMessage,
   useChannelActionContext,
   useChannelStateContext,
   useChatContext,
 } from '../../../context';
-import { Event, MessageResponse } from 'stream-chat';
-import { DefaultStreamChatGenerics } from '../../../types';
+import type { Channel, Event, MessageResponse } from 'stream-chat';
+import type { DefaultStreamChatGenerics } from '../../../types';
+
+const hasReadLastMessage = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+>(
+  channel: Channel<StreamChatGenerics>,
+  userId: string,
+) => {
+  const lastMessageIdInCurrentMessageSet = channel.state.messages.slice(-1)[0]?.id;
```

Comment:
I think this should be `channel.state.latestMessages` because in case of a message jump `channel.state.messages` can point to an older message set

Reply:
good point 👍